Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -45,15 +39,18 @@ namespace AzToolsFramework
         void InstanceDomGenerator::UnregisterInstanceDomGeneratorInterface()
         {
             m_prefabSystemComponentInterface = nullptr;
-            m_prefabFocusInterface = nullptr;
 
             AZ::Interface<InstanceDomGeneratorInterface>::Unregister(this);
         }
 
         bool InstanceDomGenerator::GenerateInstanceDom(const Instance* instance, PrefabDom& instanceDom)
         {
-            // Retrieve focused instance
-            auto focusedInstance = m_prefabFocusInterface->GetFocusedPrefabInstance(s_editorEntityContextId);
+            // Retrieve focused instance.
+            auto prefabFocusInterface = AZ::Interface<PrefabFocusInterface>::Get();
+            AZ_Assert(prefabFocusInterface, "Prefab - InstanceDomGenerator::GenerateInstanceDom - "
+                "Prefab Focus Interface couldn not be found.");
```

Comment:
I suggest keeping `Get()` here at this time. We can continue doing mroe refactoring when moving in prototype changes, but at this time it is hard to tell how dependencies would form.

For example, we are exposing _InstanceUpdateExecutor_ to _PrefabFocusHandler_; however, this dependency can be avoided if we add new corresponding APIs in _PrefabSystemComponent_ to put instances in propagation queue.